### DISCUSSION TITLE
USD Updating and exporting/converting FBX

### DISCUSSION THREAD

**Discussion by josemcasasp** [Autodesk/maya-usd#1448] [Q&A] [answered]
Hi guys, just trying to use USD as a reference system so as a team we can split the work with a better system/architecture.

Now im facing these 2 things: 

Create a usd asset.
Import my usd asset into new scene in a new stage layer.
Modify my usd asset.
**How can i update the USD asset in the scene without reopening maya?**

**What about if i want to convert the USD into maya assets?** I mean once i have the scene ready i want to combine or modify the assets and also export them as a FBX. Is it possible?

Dont know if USD will works for me maybe its not meant to be used like this?
Thanks

**Comment by wtelford** [comment, marked as answer]
Hi Jose,
I see two independent questions here.

**Question 1: If I have a USD stage in memory in Maya and I modify the USD data on disk with another program, can I easily reload the USD stage?**

Yes you can. In the Attribute Editor for the USD proxyShape you can click the circled button to reload the USD data from disk.
![image](https://user-images.githubusercontent.com/1900693/120828699-735d8200-c522-11eb-8f3e-bd3853895bb9.png)

**Question 2: Is there a way to convert in-memory USD data to Maya data?**

Yes there is a way to do in-memory data translation, but it currently requires scripting to accomplish. MayaUSD's import/export can be set up to target an in-memory layer instead of a file. You can read more about how you might accomplish that in the following GitHub discussion https://github.com/Autodesk/maya-usd/discussions/1054#discussioncomment-641407

**Comment by dgovil** [comment]
On the first point, USD itself may still have issues if you have two apps
writing to the same file at once, and you are using usdc files. (not a Maya
limitation to be clear)

Since usda files get read in all at once, they're safe to write to from
multiple apps.
usdc files are memory mapped to load on demand so you might get
corruption/crashing if an app is reading data while another writes.

Just putting that out there as a caveat.
